src/Access/Authentication.cpp

```diff
@@ -75,6 +80,19 @@ bool Authentication::isCorrectPassword(const String & password_) const
             return encodeSHA1(first_sha1) == password_hash;
         }
 
+        case LDAP_SERVER:
+        {
+            if (!external_authenticators)
+                throw Exception("External authenticators are not configured", ErrorCodes::BAD_ARGUMENTS);
```

Comment:
I think this code is made more complex that it should be. You are never going to pass nullptr as `external_authenticators` to this function, so `external_authentications` should be a reference, not a pointer.

Reply:
Well, it was a reference up until we decided to allow reloaded config to update it. After that, a rare case, when the config reloader hasn't read the config, `external_authenticators` is null.